app/models/donation.rb

```diff
@@ -91,16 +90,25 @@ def self.daily_quantities_by_source(start, stop)
                       .sum("line_items.quantity")
   end
 
-  # def self.total_received
-  #    self.includes(:line_items).map(&:total_quantity).reduce(0, :+)
-  #  end
-
-  def track(item, quantity)
-    if contains_item_id?(item.id)
-      update_quantity(quantity, item)
-    else
-      LineItem.create(itemizable: self, item_id: item.id, quantity: quantity)
+  def replace_increase!(new_donation_params)
+    old_data = to_a
+    item_ids = line_items_attributes(new_donation_params).map { |i| i[:item_id].to_i }
+
+    ActiveRecord::Base.transaction do
```

Comment:
You guys really took to ActiveRecord::Base.transaction.

Reply:
Yeah! One of the major design decisions I wanted was to make the DB tell us if quantitiies were violated rather than the methods themselves trying to "predict" if they were.

Reply:
It made such a huge difference, for real. The inventory changes are a lot more robust now.